src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStreamReadAsyncMemoryOverloadsTests.cs

```diff
@@ -578,6 +576,43 @@ public static IEnumerable<object[]> CSharpInlineByteArrayTestData()
                                         "(new byte[s.Length]).AsMemory(0, (int)s.Length), new CancellationToken()" };
         }
 
+        [Fact]
+        public Task CS_Fixer_Diagnostic_EnsureSystemNamespaceAutoAdded()
```

Comment:
@mavasani I am seeing the following error when I run any of my new unit tests:


<details>
  <summary>Error</summary>

  ```
  Message: 
      Context: Iterative code fix application
      The upper limit for the number of code fix iterations was exceeded
      Expected: True
      Actual:   False
    Stack Trace: 
      XUnitVerifier.True(Boolean assert, String message) line 62
      <FixEachAnalyzerDiagnosticAsync>d__43.MoveNext() line 538
      --- End of stack trace from previous location where exception was thrown ---
      ExceptionDispatchInfo.Throw()
      <VerifyFixAsync>d__42.MoveNext() line 434
      --- End of stack trace from previous location where exception was thrown ---
      TaskAwaiter.ThrowForNonSuccess(Task task)
      TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      ConfiguredTaskAwaiter.GetResult()
      <VerifyFixAsync>d__40.MoveNext() line 371
      --- End of stack trace from previous location where exception was thrown ---
      TaskAwaiter.ThrowForNonSuccess(Task task)
      TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      ConfiguredTaskAwaiter.GetResult()
      <RunAsync>d__38.MoveNext() line 247
      --- End of stack trace from previous location where exception was thrown ---
      TaskAwaiter.ThrowForNonSuccess(Task task)
      TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      --- End of stack trace from previous location where exception was thrown ---
      TaskAwaiter.ThrowForNonSuccess(Task task)
      TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
      --- End of stack trace from previous location where exception was thrown ---
      TaskAwaiter.ThrowForNonSuccess(Task task)
      TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
  ```
</details>

I am not sure what the root cause is. When I step into the fixer code with the debugger, and put a breakpoint right after the new root is generated, the code seems to be correct - the `System` namespace gets added where the unit tests expect it.

Any ideas what could be causing the failure?

Reply:
@Evangelink @sharwell may also know what's causing this error.

Reply:
This indicates that either the diagnostic was not resolved after the required iterations or a new diagnostic was produced. Are you able to debug into the testing library to figure out?

Reply:
Try setting the `FixOne` option, and fixing any remaining issues with the test. You'll likely see that after the code fix is applied, problems are still appearing in the code.

Reply:
Thanks, @sharwell.

Note to self - Found a `FixOne` usage example here:
[roslyn/src/EditorFeatures/CSharpTest/Diagnostics/MakeMethodSynchronous/MakeMethodSynchronousTests.cs#L458](https://github.com/dotnet/roslyn/blob/8a7bfd860e0544078ddbe6a0192f5d9ca0edac41/src/EditorFeatures/CSharpTest/Diagnostics/MakeMethodSynchronous/MakeMethodSynchronousTests.cs#L458)
```cs
            await new VerifyCS.Test
            {
                TestCode = source,
                FixedState =
                {
                    Sources = { expected },
                    MarkupHandling = MarkupMode.Allow,
                },
                CodeFixTestBehaviors = CodeFixTestBehaviors.FixOne,
            }.RunAsync();
```

Reply:
Found the root cause: I had to replace the invocation node _before_ adding the using/import.
If I add the using/import first, then it seems the reference to the original location of the invocation to update, gets lost (the code does not get properly updated), so attempting to replace the invocation fails silently.